src/NUnitFramework/tests/Constraints/EqualConstraintTests.cs

```diff
@@ -520,6 +521,90 @@ public void ErrorIfUlpsIsUsedOnDecimal()
 
         #endregion
 
+        #region ObjectsThatAreAlsoIEnumerableEquality
+
+        public sealed class ObjectsThatAreAlsoIEnumerableEquality
```

Comment:
These tests demonstrate the flip side, where clearly the object isn't equal, but because it so happens to implement IEnumerable<T>, it matches incorrectly